syncer/sharding_group.go

```diff
@@ -441,57 +493,68 @@ func (k *ShardingGroupKeeper) LeaveGroup(targetSchema, targetTable string, sourc
 //   isSharding: whether the source table is in a sharding group
 //   group: the sharding group
 //   synced: whether the source table's sharding group synced
+//   active: whether is active DDL in sequence sharding DDL
 //   remain: remain un-synced source table's count
-func (k *ShardingGroupKeeper) TrySync(targetSchema, targetTable, source string, pos, endPos mysql.Position, ddls []string) (needShardingHandle bool, group *ShardingGroup, synced bool, remain int, err error) {
-	tableID, schemaOnly := GenTableID(targetSchema, targetTable)
+func (k *ShardingGroupKeeper) TrySync(
+	targetSchema, targetTable, source string, pos, endPos mysql.Position, ddls []string) (
+	needShardingHandle bool, group *ShardingGroup, synced, active bool, remain int, err error) {
+
+	targetTableID, schemaOnly := GenTableID(targetSchema, targetTable)
 	if schemaOnly {
 		// NOTE: now we don't support syncing for schema only sharding DDL
-		return false, nil, true, 0, nil
+		return false, nil, true, false, 0, nil
 	}
 
 	k.Lock()
 	defer k.Unlock()
 
-	group, ok := k.groups[tableID]
+	group, ok := k.groups[targetTableID]
 	if !ok {
-		return false, group, true, 0, nil
+		return false, group, true, false, 0, nil
 	}
-	synced, remain, err = group.TrySync(source, pos, endPos, ddls)
-	return true, group, synced, remain, errors.Trace(err)
+	synced, active, remain, err = group.TrySync(source, pos, endPos, ddls)
+	return true, group, synced, active, remain, errors.Trace(err)
 }
 
-// InSyncing checks whether the source table is in syncing
-func (k *ShardingGroupKeeper) InSyncing(targetSchema, targetTable, source string) bool {
-	tableID, _ := GenTableID(targetSchema, targetTable)
-	k.RLock()
-	defer k.RUnlock()
-	group, ok := k.groups[tableID]
-	if !ok {
-		return false
+// CheckSyncing checks the source table syncing status
+// returns
+//   needShardingHandle: whether the source table in is a sharding group
```

Comment:
Is `needShardingHandle` matching `whether the source table in is a sharding group` well?